corehq/form_processor/models/cases.py

```diff
@@ -294,6 +294,21 @@ def hard_delete_cases(self, domain, case_ids, *, publish_changes=True):
 
         return deleted_count
 
+    def hard_delete_cases_before_cutoff(self, cutoff, dry_run=True):
```

Comment:
Should this publish changes to Kafka to ensure the cases are removed from ES?

Reply:
The cases are removed from ES when they’re soft deleted since to the user it's supposed to look like a hard deletion (won't show up in reports).